File: internal/api/model/bluesky.go
// GoToSocial
// Copyright (C) GoToSocial Authors admin@gotosocial.org
// SPDX-License-Identifier: AGPL-3.0-or-later

package model

import "time"

type BlueskyConnection struct {
	Connected         bool      `json:"connected"`
	AuthMethod        string    `json:"auth_method,omitempty"`
	DID               string    `json:"did,omitempty"`
	Handle            string    `json:"handle,omitempty"`
	ProfileURL        string    `json:"profile_url,omitempty"`
	CrosspostPublic   bool      `json:"crosspost_public"`
	ShowProfileFollow bool      `json:"show_profile_follow"`
	Configured        bool      `json:"configured"`
	Status            string    `json:"status"`
	StatusMessage     string    `json:"status_message,omitempty"`
	NeedsReconnect    bool      `json:"needs_reconnect"`
	PendingDeliveries int       `json:"pending_deliveries"`
	DeadDeliveries    int       `json:"dead_deliveries"`
	DeadNotifications int       `json:"dead_notifications"`
	LastErrorAt       time.Time `json:"last_error_at,omitempty"`
	LastSyncAt        time.Time `json:"last_sync_at,omitempty"`
}

type BlueskySettingsUpdateRequest struct {
	CrosspostPublic   *bool `form:"crosspost_public" json:"crosspost_public"`
	ShowProfileFollow *bool `form:"show_profile_follow" json:"show_profile_follow"`
}

type BlueskyConnectRequest struct {
	Identifier string `form:"identifier" json:"identifier" binding:"required"`
}

type BlueskyAppPasswordRequest struct {
	Identifier  string `form:"identifier" json:"identifier" binding:"required"`
	AppPassword string `form:"app_password" json:"app_password" binding:"required,max=256"`
}

type BlueskyConnectResponse struct {
	AuthorizationURL string `json:"authorization_url"`
}

File: web/source/settings/lib/types/bluesky.ts
/*
	GoToSocial
	Copyright (C) GoToSocial Authors admin@gotosocial.org
	SPDX-License-Identifier: AGPL-3.0-or-later

	This program is free software: you can redistribute it and/or modify
	it under the terms of the GNU Affero General Public License as published by
	the Free Software Foundation, either version 3 of the License, or
	(at your option) any later version.

	This program is distributed in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
	GNU Affero General Public License for more details.

	You should have received a copy of the GNU Affero General Public License
	along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

export interface BlueskyConnection {
	connected: boolean;
	handle?: string;
	profile_url?: string;
	crosspost_public: boolean;
	show_profile_follow: boolean;
	configured: boolean;
	pending_deliveries: number;
	dead_deliveries: number;
	dead_notifications: number;
	last_error?: string;
	last_error_at?: string;
	last_sync_at?: string;
}

export interface BlueskySettingsUpdate {
	crosspost_public: boolean;
	show_profile_follow: boolean;
}

export interface BlueskyConnectResponse {
	authorization_url: string;
}
